setup.cfg

```diff
@@ -5,6 +5,9 @@ universal=1
 description_file = README.md
 
 [flake8]
+# ref: https://flake8.pycqa.org/en/2.5.5/warnings.html#error-codes
+# E302,E305: rules that demand 2 blank lines
+extend-ignore=E302,E305
```

Comment:
remove, keep flake8 rules consistent (e.g. with `nbcelltests`). You can use `autopep8` to auto apply this, it will apply this rule without you manually having to go and insert the newlines

Reply:
I completely loathe these styling rules (similarly, I suspect, to how you feel about the default `max-line-length=79`). I would much prefer if they were not enforced across the codebase.

If consistency is the issue, I'd be glad to also create a PR disabling these rules on `nbcelltests` as well.

Reply:
which one in particular? I like 2 newlines in top level blocks (305), not sure how it differs from E302.

Reply:
Here's the exact rule definitions (from https://pycodestyle.readthedocs.io/en/latest/intro.html#error-codes):

> E302 | expected 2 blank lines, found 0
> E305 | expected 2 blank lines after end of function or class

E305 seems to be for exactly what it says (2 blank lines after class or function definitions), while E302 seems to cover all other cases (eg 2 blank lines before definitions).

Reply:
I'm fine with sticking to "2 blank lines before and after function definitions" as a rule of thumb, I just also think that there are a great number of exceptions to said rule that improve the readability and flow of code. For example, if I reenable E305, the only thing it currently complains about is this block:

https://github.com/jpmorganchase/jupyter-fs/blob/3b527bf8fa5c2ff399b575d2dc5a344264258d74/setup.py#L59-L65

This kind of construct is the closest thing syntactically to multi-line anonymous functions in Python (ie a function that is used exactly once at its place of definition). I'd like to be able to keep the line in which `runPackLabextension` is used enjambed with the definition of `runPackLabextension`.

Reply:
On the other hand, if I reenable E302, one of the things it complains about is that there's only one space in between the definitions of these closely related low-level S3 handling functions:

https://github.com/jpmorganchase/jupyter-fs/blob/3b527bf8fa5c2ff399b575d2dc5a344264258d74/jupyterfs/tests/test_pyfilesystem_manager.py#L38-L76

I think it makes sense to have two spaces before and after such a block of function definitions (which there currently are), but using only one space spaces in between each definition is a useful way of expressing their relatedness.

Reply:
@telamonian You can set up a git filter action to auto-format back and forth between different styles ;)